Stripe/STPPaymentContext.swift

```diff
@@ -0,0 +1,1129 @@
+  ///   - completion:     Call this block when you're done creating a payment intent (or subscription, etc) on your backend. If it succeeded, call `completion(STPPaymentStatusSuccess, nil)`. If it failed with an error, call `completion(STPPaymentStatusError, error)`. If the user canceled, call `completion(STPPaymentStatusUserCancellation, nil)`.
+  func paymentContext(
+    _ paymentContext: STPPaymentContext,
+    didCreatePaymentResult paymentResult: STPPaymentResult,
+    completion: @escaping STPPaymentStatusBlock
+  )
+  /// This is invoked by an `STPPaymentContext` when it is finished. This will be called after the payment is done and all necessary UI has been dismissed. You should inspect the returned `status` and behave appropriately. For example: if it's `STPPaymentStatusSuccess`, show the user a receipt. If it's `STPPaymentStatusError`, inform the user of the error. If it's `STPPaymentStatusUserCancellation`, do nothing.
+  /// - Parameters:
+  ///   - paymentContext: The payment context that finished
+  ///   - status:         The status of the payment - `STPPaymentStatusSuccess` if it succeeded, `STPPaymentStatusError` if it failed with an error (in which case the `error` parameter will be non-nil), `STPPaymentStatusUserCancellation` if the user canceled the payment.
+  ///   - error:          An error that occurred, if any.
+  func paymentContext(
+    _ paymentContext: STPPaymentContext,
+    didFinishWith status: STPPaymentStatus,
+    error: Error?
+  )
+
+  /// Inside this method, you should verify that you can ship to the given address.
+  /// You should call the completion block with the results of your validation
+  /// and the available shipping methods for the given address. If you don't implement
+  /// this method, the user won't be prompted to select a shipping method and all
+  /// addresses will be valid. If you call the completion block with nil or an
+  /// empty array of shipping methods, the user won't be prompted to select a
+  /// shipping method.
+  /// @note If a user updates their shipping address within the Apple Pay dialog,
+  /// this address will be anonymized. For example, in the US, it will only include the
+  /// city, state, and zip code. The payment context will have the user's complete
+  /// shipping address by the time `paymentContext:didFinishWithStatus:error` is
+  /// called.
+  /// - Parameters:
+  ///   - paymentContext:  The context that updated its shipping address
+  ///   - address: The current shipping address
+  ///   - completion:      Call this block when you're done validating the shipping
+  /// address and calculating available shipping methods. If you call the completion
+  /// block with nil or an empty array of shipping methods, the user won't be prompted
+  /// to select a shipping method.
+  @objc optional func paymentContext(
```

Comment:
`@objc` is unnecessary here right?

Reply:
I feel like it's implied, but the compiler thinks it's needed because all optional protocol functions must be public.